src/main/java/at/hannibal2/skyhanni/features/misc/GeorgeDisplay.kt

```diff
@@ -0,0 +1,132 @@
+                val (cheapestTier, petPrice) = findCheapestTier(pet, originalTier)
+                val displayPetString = "${LorenzRarity.getById(cheapestTier)?.formattedName} $origPetString"
+                if (petPrice != -1.0) {
+                    totalCost += petPrice
+                    this@buildList.add(
+                        Renderable.clickAndHover(
+                            text = " §7- $displayPetString§7: §6${petPrice.addSeparators()} coins",
+                            tips = listOf(
+                                "§aClick to run §e/ahs ] $origPetString §ato find it on the Auction House.",
+                                "§aNotes: §eSet the rarity filter yourself. §cBooster Cookie required!"
+                            ),
+                            onClick = { LorenzUtils.sendCommandToServer("ahs ] $origPetString") }
+                        ))
+                } else {
+                    val typeOfWiki = if (useFandomWiki) "Fandom" else "Official Hypixel"
+                    this@buildList.add(
+                        Renderable.clickAndHover(
+                            text = " §7- $displayPetString§7: §eNot on AH; view its $typeOfWiki article here.",
+                            tips = listOf("§eClick to open the $typeOfWiki Wiki article for the $displayPetString §eto view how to obtain it."),
+                            onClick = {
+                                val urlCompliantPet = displayPetString.removeColor().replace(" ", "%20")
+                                if (useFandomWiki) OSUtils.openBrowser("https://hypixel-skyblock.fandom.com/wiki/Special:Search?query=$urlCompliantPet&scope=internal")
+                                else OSUtils.openBrowser("https://wiki.hypixel.net/index.php?search=$urlCompliantPet")
+                            }
+                        ))
+                }
+            }
+        this.add(Renderable.string("§dTotal cost §7(§6Lowest BIN§7): §6${totalCost.addSeparators()} coins"))
+        if (config.otherRarities) this.add(Renderable.string("§c§lDisclaimer:§r§c Total does not include costs to upgrade via Kat."))
+    }
+
+    private fun findCheapestTier(pet: String, originalTier: Int) = buildList<Pair<Int, Double>> {
+        this.add(originalTier to petInternalName(pet, originalTier).getPetPrice())
+        if (config.otherRarities) {
+            this.add(originalTier - 1 to petInternalName(pet, originalTier - 1).getPetPrice(otherRarity = true))
+            if (originalTier != 5) this.add(
+                originalTier + 1 to petInternalName(pet, originalTier + 1).getPetPrice(
+                    otherRarity = true
+                )
+            )
+        }
+    }.minBy { it.second }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.ChestGuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (!offerPetsChestPattern.matches(InventoryUtils.openInventoryName())) return
```

Comment:
Check this once on inventory open and use a boolean inInventory, instead of checking the regex once per render